Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: Add -Wa,-mimplicit-it=thumb to CFLAGS if building on ARM
 The build of the nvidia-drm kernel module fails on kernel 4.5+ on armhf. The
 following error is reported:
   /tmp/ccy8zHSc.s: Assembler messages:
   /tmp/ccy8zHSc.s:97: Error: thumb conditional instruction should be in IT
     block -- `movne r0,#1'
 As a workaround, add -Wa,-mimplicit-it=thumb to the CFLAGS if building on arm.

--- a/conftest.sh
+++ b/conftest.sh
@@ -146,7 +146,7 @@
     done
 
     if [ "$ARCH" = "arm" ]; then
-        MACH_CFLAGS="$MACH_CFLAGS -D__LINUX_ARM_ARCH__=7"
+        MACH_CFLAGS="$MACH_CFLAGS -D__LINUX_ARM_ARCH__=7 -Wa,-mimplicit-it=thumb"
     fi
 
     # Add the mach-default includes (only found on x86/older kernels)
